platform/ext/target/nordic_nrf/common/native_drivers/spu.c

```diff
@@ -158,6 +161,66 @@ void spu_regions_flash_config_non_secure_callable(uint32_t start_addr,
 		1 /* Lock */);
 }
 
+uint32_t spu_regions_flash_get_base_address_in_region(uint32_t region_id)
+{
+	return FLASH_BASE_ADDRESS +
+		((region_id - FLASH_SECURE_ATTRIBUTION_REGIONS_START_ID) *
+			FLASH_SECURE_ATTRIBUTION_REGION_SIZE);
+}
```

Comment:
How about merging flash and ram to something like this?
```suggestion
uint32_t spu_get_base_address_in_region(uint32_t region_id)
{
	if ((base_spu_id >= spu_regions_flash_get_start_id()) &&
		(last_spu_id <= spu_regions_flash_get_last_id())) {
		return FLASH_BASE_ADDRESS +
			((region_id - FLASH_SECURE_ATTRIBUTION_REGIONS_START_ID) *
				FLASH_SECURE_ATTRIBUTION_REGION_SIZE);
	} else if ((base_spu_id >= spu_regions_sram_get_start_id()) &&
		(last_spu_id <= spu_regions_sram_get_last_id())) {	
		return SRAM_BASE_ADDRESS +
			((region_id - SRAM_SECURE_ATTRIBUTION_REGIONS_START_ID) *
				SRAM_SECURE_ATTRIBUTION_REGION_SIZE);
	}
	return 0;
}
```
Likewise for `spu_get_last_address_in_region()` and `spu_get_region_size()`

It would mean you can avoid duplication in `tfm_spm_hal_has_access_to_region()`.